tests/unit/ml_handlers/test_huggingface.py

```diff
@@ -174,4 +174,42 @@ def test_hf_translation(self, mock_handler):
         '''
         self.hf_test_run(mock_handler, model_name, create_sql, predict_sql)
 
+    @patch('mindsdb.integrations.handlers.postgres_handler.Handler')
+    def test_hf_text2text(self, mock_handler):
+
+        # create predictor
+        create_sql = '''                
+        CREATE MODEL mindsdb.text_generator
+        predict PREDICTION
+        USING
+            engine='huggingface',
+            join_learn_process=true,
+            task = "text2text-generation",
+            model_name = "google/flan-t5-base",
+            input_column = "text_short"
+        '''
+
+        model_name = 'text_generator'
+
+        predict_sql = '''
```

Comment:
Does it make sense to run the text generator on the input defined for `pg.df` at the start of this file? Perhaps you can change this into the `SELECT * FROM ... WHERE` query you've left as a comment in line 214, instead.